frontend/src/core/codemirror/cm.ts

```diff
@@ -94,17 +95,20 @@ export const setupCodeMirror = (opts: CodeMirrorSetupOpts): Extension[] => {
   ];
 };
 
-const startCompletionAtEndOfLine = (cm: EditorView): boolean => {
+const insertTabOrComplete = (cm: EditorView): boolean => {
```

Comment:
This starts a completion only if the previous character is not whitespace. Otherwise it inserts a tab.

However, when testing, it seems like sometimes one or two spaces are inserted instead of the full indentation unit, which we set to four spaces. Not sure why ...